src/Sylius/Bundle/ApiBundle/DataTransformer/ChannelPricingInputDataTransformer.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\DataTransformer;
+
+use ApiPlatform\Core\DataTransformer\DataTransformerInterface;
+use ApiPlatform\Core\Serializer\AbstractItemNormalizer;
+use Sylius\Bundle\ApiBundle\DTO\ChannelPricingInput;
+use Sylius\Bundle\ApiBundle\Helpers\PriceTransformer;
+use Sylius\Component\Core\Model\ChannelPricing;
+
+class ChannelPricingInputDataTransformer implements DataTransformerInterface
+{
+    /**
+     * @param object $input
+     * @param string $to
+     * @param array $context
+     * @return object|ChannelPricing
+     */
+    public function transform($input, string $to, array $context = [])
+    {
+        if (get_class($input) !== ChannelPricingInput::class) {
+            throw new \Exception('the $input parameter should be instance of ' . ChannelPricingInput::class);
+        }
```

Comment:
Here I see two incorrect points.
The first one is why are you getting the class name of object, and then use it with `!==` If we need to check that an object is instance of some interface of class we may use `instanceof` 

```suggestion
        if (!$input instanceof ChannelPricingInput) {
            throw new \Exception('the $input parameter should be instance of ' . ChannelPricingInput::class);
        }
```

The second one point is, why are you validating this object on transformation method? DataTransformerInterface says https://i.imgur.com/RQJHMBY.png so we have to do all validations about supportsTransformation in the right method.

Reply:
I repeat myself so I'll say **Attention please!**
This functional is supposed to work ,also, when other resources than ChannelPricing are requested;
For instance when creating a ProductVariant, you can add some channelPricings to it, and I need to convert those little channel-pricings to; 
1. Using that long requirement statement you've suggested below will jump over channel-pricing's transformation.
2. I have changed the default denormalizer for this DTO (ChannelPricingINputDenormalizer), what denormalize() will return that will be the object;
   - you can't check what's gonna be the instance of the received object inside of supportsTransformation because that happens prior to the actual transformation;
3. I use get_class instead of `instanceof` because I need specifically that class;